src/common/commonutils/FileUtils.c

```diff
@@ -81,7 +81,7 @@ static bool SaveToFile(const char* fileName, const char* mode, const char* paylo
             {
                 for (i = 0; i < payloadSizeBytes; i++)
                 {
-                    if (payload[i] != fputc(payload[i], file))
+                    if ((unsigned char)payload[i] != fputc(payload[i], file))

```

Comment:
>fputc [](http://example.com/codeflow?start=53&length=5)

fputc returns int. Signed. Why cats to unsigned the payload byte being compared? Please explain or revert.

Reply:
https://man7.org/linux/man-pages/man3/fgetc.3.html
> fgetc() reads the next character from stream and returns it as an
       unsigned char cast to an int, or EOF on end of file or error.

Reply:
Then it shouldn't it be like 
```
  if (payload[i] != (unsigned char)fputc(payload[i], file))
```